packages/react/src/table/TableHeaderSortable.tsx

```diff
@@ -0,0 +1,77 @@
+import { Box, Flex } from '../box';
+import { BaseButton } from '../button';
+import { boxPalette } from '../core';
+import { ArrowDownIcon, ArrowUpIcon } from '../icon';
+import { TableHeaderProps } from './TableHeader';
+
+export type TableSortDirection = 'ASC' | 'DESC' | undefined;
+
+type TableHeaderSortableProps = TableHeaderProps & {
+	sort: TableSortDirection;
+	onClick?: () => void;
+};
+
+/** Used in place of TableHeader when users can click a column header to
+ * sort the table. */
+export const TableHeaderSortable = ({
+	children,
+	textAlign = 'left',
+	width,
+	sort,
+	onClick,
+}: TableHeaderSortableProps) => {
+	const Icon = sort === 'DESC' ? ArrowUpIcon : ArrowDownIcon;
+
+	return (
+		<Box
+			as="th"
+			scope="col"
+			aria-sort={getSortLabel(sort)}
+			focus
+			width={width}
+			css={{
+				textAlign,
+				borderBottomColor: boxPalette.foregroundAction,
+			}}
+			{...(sort && {
+				borderBottom: true,
+				borderBottomWidth: 'xl',
+			})}
+		>
+			<Flex
+				as={BaseButton}
+				gap={0.5}
+				onClick={onClick}
+				type="button"
+				padding={0.75}
+				fontWeight="bold"
+				width="100%"
+				justifyContent="space-between"
+				alignItems="center"
+				focus
+				css={{
+					textDecoration: 'underline',
+					svg: {
+						display: sort ? 'block' : 'none',
+					},
+					'&:hover': {
+						backgroundColor: boxPalette.backgroundShade,
+						svg: {
+							display: 'block',
```

Comment:
This isn't required because you. have defined display block on line 55

Reply:
On 55, display=block is defined to svgs only if their column is the active sort.
Here on 55 I ensure that the icon is alway visible on hover, regardless of if the column is the active sort